src/algorithms/planners/GeoPlanner.h

```diff
@@ -0,0 +1,330 @@
+
+                // Set minimum travel score from incoming packet.
+                if (stPacket.vData.size() > 0)
+                {
+                    m_dBeta = static_cast<double>(stPacket.vData[0]);
+                    // Submit logger message.
+                    LOG_NOTICE(logging::g_qSharedLogger, "Incoming Packet: Setting GeoPlanner beta bias to {}", this->m_dBeta);
+                }
+            };
+
+            ////////////////////////////////////
+            // Private member variables.
+            ////////////////////////////////////
+
+            int m_nStartID, m_nEndID;                                        // Unique identifiers for the start and end points.
+            double m_dBeta;                                                  // Bias factor for travel scores.
+            double m_dMinTravScore;                                          // Minimum travel score threshold for path planning.
+            double m_dTileSize;                                              // Size of the grid tiles in meters.
+            double m_dSearchRadius;                                          // Search radius for finding neighbors.
+            double m_dMaxSearchTimeSeconds;                                  // The maximum time to spend searching for a path in seconds.
+            LiDARHandler* m_pLiDARHandler;                                   // Pointer to the LiDARHandler instance for fetching geospatial data.
+            std::unique_ptr<logging::graphing::PathTracer> m_pPathTracer;    // Path tracer for 3D visualization.
+            std::unique_ptr<KDTree2D> m_pKDTree;                             // KD-tree for fast spatial queries over loaded tiles.
+            std::mutex m_muPathGenMutex;                                     // Mutex to protect path planning operations.
+
+            // AStar's algorithm related variables.
+
+            /*
+             * The core of AStar's algorithm is a priority queue (min-heap) that stores the open set of nodes to be evaluated.
+             * The priority queue is ordered by the estimated total cost (f = g + h) of reaching the goal from the start node
+             * through each node. We also maintain a map to track the best path to each node (predecessors) and a set of
+             * nodes that have already been evaluated (closedSet).
+             */
+            std::priority_queue<PlannerState, std::vector<PlannerState>, PlannerStateCompare> m_pqOpenSetNextBest;    // Priority queue (min-heap) to be evaled.
+            std::unordered_map<int, int> m_umPredecessors;                                                            // Maps point IDs to their predecessor's ID.
+            std::unordered_set<int> m_usClosedSet;                                                                    // Set of point IDs that have been evaluated.
+            std::unordered_map<int, PlannerState> m_umAllStates;    // Maps point IDs to their corresponding PlannerState.
```

Comment:
Align comment block with previous 3 lines if possible for cleaner reading

```suggestion
            std::unordered_map<int, PlannerState> m_umAllStates;                                                      // Maps point IDs to their corresponding PlannerState.
```